components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5541,18 +5542,31 @@ private Attribute getSchemaAttributes(ExternalClaim scimClaim, LocalClaim mapped
                                           boolean isExtensionAttr) {
 
         String name = scimClaim.getClaimURI();
+        String claimDielectURI = scimClaim.getClaimDialectURI();
+        boolean isCustomSchemaAttr = StringUtils.equalsIgnoreCase(claimDielectURI,getCustomSchemaURI());
+        boolean isComplexInAdditionalProp = false;
```

Comment:
```suggestion
        boolean isComplexAttribute = false;
```

Reply:
This variable is dedicated to check if the attribute is defined as complex in additional properties.

Reply:
For Scim custom attributes, if additionalProperty contains a key with DATA_TYPE and it's value is COMPLEX means the attribute is complex.

Now your intention is to find whether the custom attribute is complex or not right?
`isComplexCustomAttribute` would be a better name

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/459/commits/a227f2e3af8b1ad2afaaab992d44c5f8e2b64c70